docs/notebooks/.ipynb_checkpoints/WMD_tutorial-checkpoint.ipynb

```diff
@@ -0,0 +1,601 @@
+    "\n",
+    "Note that normalizing the vectors can take some time, especially if you have a large vocabulary and/or large vectors.\n",
+    "\n",
+    "Usage is illustrated in the example below. It just so happens that the vectors we have downloaded are already normalized, so it won't do any difference in this case."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 44,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "Cell took 11.987278 seconds to run.\n"
+     ]
+    }
+   ],
+   "source": [
+    "# Normalizing word2vec vectors.\n",
+    "start = time()\n",
+    "\n",
+    "model.init_sims(replace=True)  # Normalizes the vectors in the word2vec class.\n",
+    "\n",
+    "distance = model.wmdistance(sentence1, sentence2)  # Compute WMD as normal.\n",
+    "\n",
+    "print 'Cell took %f seconds to run.' %(time() - start)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Similarity queries\n",
+    "\n",
+    "You can use WMD to get the most similar documents to a query, using the `WmdSimilarity` class. Its interface is similar to what is described in the [Similarity Queries](https://radimrehurek.com/gensim/tut3.html) Gensim tutorial.\n",
+    "\n",
+    "> **Important note:**\n",
+    ">\n",
+    "> WMD is a measure of *distance*. The similarities in `WmdSimilarity` are simply the *negative distance*. Be careful not to confuse distances and similarities.\n",
+    "\n",
+    "Let's try similarity queries using some real world data. For that we'll be using Yelp reviews, available at http://www.yelp.com/dataset_challenge. Specifically, we will be using reviews of a single restaurant.\n",
+    "\n",
+    "This time around, we are going to train the Word2Vec embeddings on the data ourselves. One restaurant is not enough to train Word2Vec properly, so we use 6 restaurants for that, but only run queries against one of them. More information about the data is in the **Yelp data** section.\n",
+    "\n",
+    "Below a JSON file with Yelp reviews is read line by line, the text is extracted, tokenized, and stopwords and punctuation are removed."
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 45,
+   "metadata": {
+    "collapsed": false
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "[nltk_data] Downloading package punkt to /home/olavur/nltk_data...\n",
+      "[nltk_data]   Package punkt is already up-to-date!\n",
+      "\n",
+      "Cell took 38.429271 seconds to run.\n"
+     ]
+    }
+   ],
+   "source": [
+    "start = time()\n",
+    "\n",
+    "import json\n",
+    "from nltk import word_tokenize\n",
+    "download('punkt')  # Download data for tokenizer.\n",
+    "\n",
+    "# Business IDs of some restaurants.\n",
+    "ids = ['4bEjOyTaDG24SY5TxsaUNQ', '2e2e7WgqU1BnpxmQL5jbfw', 'zt1TpTuJ6y9n551sw9TaEg',\n",
```

Comment:
what are these restaurants?
link to their yelp http://en.yelp.be/biz/mon-ami-gabi-las-vegas-2